File: TriviaQuiz/app/src/main/java/com/example/triviaquiz/MainActivity.java
package com.example.triviaquiz;

import static com.example.triviaquiz.RegisterationActivity.SAVEDATA;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.splashscreen.SplashScreen;

import android.content.Intent;
import android.os.Bundle;
import android.os.UserHandle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.Objects;


public class MainActivity extends AppCompatActivity {
    SharedPreference sharedPreference;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        SplashScreen.installSplashScreen(this);

        setContentView(R.layout.activity_main);

        sharedPreference =   new SharedPreference(this);

        // if there is user in shared preference then automatically go to QuizhistoryActivity

        if (sharedPreference.getToken() != null) {
            Intent intent = new Intent(MainActivity.this, QuizHistoryActivity.class);
            startActivity(intent);
            finish();


        }

        TextView signUpButton = findViewById(R.id.btnSignUp);
        Button signInButton = findViewById(R.id.button_signIn);

        /**
         *       This button is   go to sign up Activity
         */

        signUpButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(MainActivity.this, RegisterationActivity.class);
                startActivity(intent);

            }
        });

        /**
         *       This button is go to sign up Activity
         */

        signInButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                Intent intent = new Intent(MainActivity.this, LoginActivity.class);
                startActivity(intent);


            }
        });


    }
}

File: NotesAppUsingViewModel/app/src/main/java/com/example/notesappusingviewmodel/MainActivity.kt
package com.example.notesappusingviewmodel

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.fragment.app.add
import androidx.fragment.app.commit
import com.example.notesappusingviewmodel.listnotes.ListNotes

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        if (savedInstanceState == null) {
            supportFragmentManager.commit {
                setReorderingAllowed(true)
                add<ListNotes>(R.id.fragmentContainer, "main")
                addToBackStack(null)
            }
        }
    }
}